packages/network/src/logic/tracker/Tracker.ts

```diff
@@ -232,15 +232,29 @@ export class Tracker extends EventEmitter {
         }
         if (this.overlayPerStreamPart[streamPartId]) {
             const instructions = this.overlayPerStreamPart[streamPartId].formInstructions(node, forceGenerate)
-            Object.entries(instructions).forEach(async ([nodeId, newNeighbors]) => {
-                const counterValue = this.instructionCounter.setOrIncrement(nodeId, streamPartId)
-                await this.instructionSender.addInstruction({
-                    nodeId,
+
+            // Send empty instruction if and only if the node is alone in the topology
+            if (this.overlayPerStreamPart[streamPartId].hasNode(node)
+                && this.overlayPerStreamPart[streamPartId].getNumberOfNodes() === 1
```

Comment:
Is this check here just in case or have there been observed cases of no neighbors being assigned after `formInstructions` call even in the presence of other nodes?

Edit: Now that I think of it, this is a possible outcome at least if node degree parameter was set to be 1. Let's say the topology already contained 2 nodes that were connected to each other. If a third node now joined, it would have no neighbors to connect to (as we wouldn't cut the existing connection in this case.) But I still wonder if this can happen with reasonable node degrees, e.g. `d=4`?